thunder/executors/torchex.py

```diff
@@ -249,6 +249,24 @@ def _arange_transform(
     return arange(start=start, step=step, end=end, device=torch_device, dtype=torch_dtype)
 
 
+def _get_grad_transform(a: TensorProxy) -> TensorProxy:
+    return prims.get_grad(a)
+
+
+def _put_grad_transform(a: TensorProxy, val: TensorProxy) -> None:
+    # what a hack. _transform_for_operator_executor_execution needs that prims are executable
+    # by some executor, and if they are not, it will descend to the subsymbols.  So, we
+    # force that and recurse so that the symbol is not removed/no error is raised.
+    # This is breaking the intent of transform_for_operator_executor_execution, in that
+    # it is not executable.  But there will be postprocessing that will make it executable.
+    prims.put_grad(a, val)
```

Comment:
Or this could be valid PyTorch code: `a.grad = val`. It would be represented with a new operator `torchex.put_grad` with an implementation function as in `torchex.shape`:
https://github.com/Lightning-AI/lightning-thunder/blob/cfcc77b1826d59e125a42de34689dc050df6c6da/thunder/executors/torchex.py#L2335-L2340
And the splitter code then would use these new get_grad and put_grad instead of the ones from the prims namespace. Still not as clean because of duplicated operators. Another option to try is to specify `python_impl` for prims.put_grad and prims.get_grad.